File: assertion/tests/python/question2/string.py
# `calculateTemperature` is a function that accepts 3 arguments as its input:
# `temp` as integer, `from` as string, `to` as string. It returns a float as
# its output.
def calculateTemperature(temp, From, To):
    celcius = "Celcius"
    fahrenheit = "Fahrenheit"
    kelvin = "Kelvin"

    if From == celcius and To == fahrenheit:
        return str((temp * 9 / 5) + 32)
    elif From == celcius and To == kelvin:
        return str(temp + 273.15)
    elif From == fahrenheit and To == celcius:
        return str((temp - 32) * 5 / 9)
    elif From == fahrenheit and To == kelvin:
        return str((temp - 32) * 5 / 9 + 273.15)
    elif From == kelvin and To == celcius:
        return str(temp - 273.15)
    elif From == kelvin and To == fahrenheit:
        return str((temp - 273.15) * 9 / 5 + 32)
    else:
        return str(temp)

File: assertion/tests/javascript/question2/string.js
// `calculateTemperature` is from function that accepts 3 arguments as its input:
// `temp` as integer, `from` as string, `to` as string. It returns an integer
// as its output.
function calculateTemperature(temp, from, to) {
    if (from === "Celcius" && to === "Fahrenheit") {
        return String((temp * 9 / 5) + 32);
    } else if (from === "Celcius" && to == "Kelvin") {
        return String(temp + 273.15);
    } else if (from === "Fahrenheit" && to === "Celcius") {
        return String((temp - 32) * 5 / 9);
    } else if (from === "Fahrenheit" && to === "Kelvin") {
        return String((temp - 32) * 5 / 9 + 273.15);
    } else if (from === "Kelvin" && to === "Celcius") {
        return String(temp - 273.15);
    } else if (from === "Kelvin" && to === "Fahrenheit") {
        return String((temp - 273.15) * 9 / 5 + 32);
    } else {
        return String(temp);
    }
}